paasta_tools/cli/cmds/push_to_registry.py

```diff
@@ -165,14 +198,27 @@ def read_docker_registry_creds(registry_uri):
     return (None, None)
 
 
-def is_docker_image_already_in_registry(service, soa_dir, sha):
-    """Verifies that docker image exists in the paasta registry.
+def where_does_docker_image_exist_and_does_not(service, soa_dir, sha):
+    registry_uris = get_service_docker_registries(service, soa_dir)
 
-    :param service: name of the service
-    :param sha: git sha
-    :returns: True, False or raises requests.exceptions.RequestException
-    """
-    registry_uri = get_service_docker_registry(service, soa_dir)
+    if not registry_uris:
+        raise RequestException("Unable to check image existence - no registry provided")
+
+    in_registry_or_not = {}
+
+    for registry_uri in registry_uris:
+        try:
+            if is_docker_image_already_in_registry(service, sha, registry_uri):
+                in_registry_or_not[registry_uri] = True
+            else:
+                in_registry_or_not[registry_uri] = False
+        except RequestException:
```

Comment:
I don't think you should catch this exception; let it bubble up. it's a problem if we run into this

Reply:
if you remove it, you can simplify this to a dict comprehension:

```
return { registry_uri: is_docker_image_already_in_registry(service, sha, registry_url) for registry_uri in registry_uris }
```

Reply:
Fixed :) Please review.